testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -919,6 +922,55 @@ def check_operator(op, method, level):
                 yield check_operator, op, method, level
 
 
+class TestGroupHash(object):
+    """
+    Groups should be hashable.
+
+    See issue #1397
+    """
+    def test_hash_exists(self):
+        def _hash_type(group):
+            assert_(isinstance(hash(group), int))
+
+        u = make_Universe(size=(3, 3, 3))
+        for level in ('atoms', 'residues', 'segments'):
+            group = getattr(u, level)
+            yield _hash_type, group
+
+    def test_hash_equality(self):
+        def _hash_equal(a, b):
+            assert_equal(hash(a), hash(b))
+
+        u = make_Universe(size=(3, 3, 3))
+        for level in ('atoms', 'residues', 'segments'):
+            a = getattr(u, level)[0:-1]
+            b = getattr(u, level)[0:-1]
+            yield _hash_equal, a, b
+
+    def test_hash_difference(self):
+        def _hash_not_equal(a, b):
+            assert_(hash(a) != hash(b))
+
+        u = make_Universe(size=(3, 3, 3))
+        for level in ('atoms', 'residues', 'segments'):
+            a = getattr(u, level)[:-1]
+            b = getattr(u, level)[1:]
+            yield _hash_not_equal, a, b
+
+    def test_hash_difference_cross(self):
+        def _hash_not_equal(a, b):
+            assert_(hash(a) != hash(b))
+
+        u = make_Universe(size=(3, 3, 3))
+        levels = ('atoms', 'residues', 'segments')
+        for level_a in levels:
```

Comment:
I spent some time today removing itertools use from cython function. It obviously messed up with my brain.